### PR TITLE
Fix init of Input CustomSerDe of function

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation

Right now, if topic has provided CustomSerDe for input param. still Instance tries to initialize it with DefaultSerDe and fails it argument doesn't supported by DefaultSerDe.

### Modifications

- Ignore initializing SerDe if CustomSerDe is provided for topic.
- Fix test cases.

### Result

CustomSerDe will work for input param.

### CODE REVIEW

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java [hunk 1]
```diff
@@ -723,7 +723,10 @@ private void setupSerDe(Class<?>[] typeArgs, ClassLoader clsLoader) {
         this.inputSerDe = new HashMap<>();
         instanceConfig.getFunctionConfig().getCustomSerdeInputsMap().forEach((k, v) -> this.inputSerDe.put(k, initializeSerDe(v, clsLoader, typeArgs, true)));
         for (String topicName : instanceConfig.getFunctionConfig().getInputsList()) {
-            this.inputSerDe.put(topicName, initializeDefaultSerDe(typeArgs, true));
+            // initialize Default-SerDe if custom-SerDe is not attached to the topic
```
Comment (srkukarni) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java]: Isn't a better check to verify that the same topic is not in both inputs and customSerdeInput?

Comment (rdhabalia) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java]: > Isn't a better check to verify that the same topic is not in both inputs and customSerdeInput?

Please correct if I am wrong. But inputs is filled by CustomSerde-map only at line#724. So, if CustomSerDe is provided and if we still try to init with DefaultSerDe then we see below exception. 

```
Caused by: java.lang.RuntimeException: Default Serializer does not support class org.apache.pulsar.functions.instance.JavaInstanceRunnableTest$TestData
	at org.apache.pulsar.functions.instance.JavaInstanceRunnable.initializeDefaultSerDe(JavaInstanceRunnable.java:711)
	at org.apache.pulsar.functions.instance.JavaInstanceRunnable.setupSerDe(JavaInstanceRunnable.java:729)
```

Comment (srkukarni) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java]: Line 724 is going over the customSerDeInput List and filling up the inputSerDe map, while line 726 goes over input List to fill the inputSerDe map. There are three cases
1) only inputs with custom serde are specified. In which case 726 is a noop
2) only inputs with default serde are specified. In which case 724 is noop
3) Function takes some inputs specified by custom serdes and one inputs using defaultserde. In this case we need to populate the inputSerDe for those topics which are going to be read using the default serializer.

**Comment by sijie** [comment]
@rdhabalia are you still working on this change? shall we move this to 2.1?

**Comment by sijie** [comment]
@rdhabalia @srkukarni I moved this to 2.1, feel free to move it back to 2.0

**Comment by sijie** [comment]
@rdhabalia is this still needed?

**Comment by rdhabalia** [comment]
As a user, I felt it was a bug because we can pass

**Pull request closed without merging**